### PR TITLE
flow: Add MongoDB prometheus exporter component

### LINKED ISSUE #2315: Flow: Add mongodb_exporter
Create `prometheus.integration.mongodb_exporter` component

### PR DESCRIPTION
**Pull request by marctc**
#### PR Description

This PR adds the ability to configure Flow to get metrics from MongoDB cluster.

#### Which issue(s) this PR fixes

Fixes #2315 

#### PR Checklist

- [x] CHANGELOG updated
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

File: docs/sources/flow/reference/components/prometheus.exporter.mongodb.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+// Configure a prometheus.scrape component to collect MongoDB metrics.
+prometheus.scrape "demo" {
+  targets    = prometheus.exporter.mongodb.example.targets
+  forward_to = [ prometheus.remote_write.default.receiver ]
+}
+
+// Configure a prometheus.relabel component to rewrite the metrics' label set.
+prometheus.relabel "mongodb_relabel" {
```
Comment (marctc) [inline docs/sources/flow/reference/components/prometheus.exporter.mongodb.md:99]: This is taken from https://grafana.com/docs/agent/latest/static/configuration/integrations/mongodb_exporter-config/. Let me know if is still valid cc @gaantunes

**Review by jcreixell — APPROVED** [review]

File: docs/sources/flow/reference/components/prometheus.exporter.mongodb.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+# injecting special characters.
+
+title: prometheus.exporter.mongodb
+---
+
+# prometheus.exporter.mongodb
+The `prometheus.exporter.mongodb` component embeds percona's [`mongodb_exporter`](https://github.com/percona/mongodb_exporter).
+
+{{% admonition type="note" %}}
+In order for this integration to work properly, you have to connect each node of your mongoDB cluster to an agent instance.
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/prometheus.exporter.mongodb.md:16]:
```suggestion
For this integration to work properly, you must have connect each node of your MongoDB cluster to an agent instance.
```

File: docs/sources/flow/reference/components/prometheus.exporter.mongodb.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+That's because this exporter does not collect metrics from multiple nodes.
+
+Additionally, you need to define two custom label for your metrics using `relabel_configs`.
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/prometheus.exporter.mongodb.md:19]:
```suggestion
You must also define two custom labels for your metrics using `relabel_configs`.
```

File: docs/sources/flow/reference/components/prometheus.exporter.mongodb.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+The first one is `service_name`, which is how you identify this node in your cluster (example: `ReplicaSet1-Node1`).
+
+The second one is `mongodb_cluster`, which is the name of your mongodb cluster, and must be set the same value for all nodes composing the cluster (example: `prod-cluster`).
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/prometheus.exporter.mongodb.md:23]:
```suggestion
The first label is `service_name`, which is how you identify this node in your cluster, for example: `ReplicaSet1-Node1`.

The second label is `mongodb_cluster`, which is the name of your MongoDB cluster. It must be set to the same value for all nodes composing the cluster, for example: `prod-cluster`.
```

File: docs/sources/flow/reference/components/prometheus.exporter.mongodb.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+{{% /admonition %}}
+
+We strongly recommend that you configure a separate user for the Agent, and give it only the strictly mandatory
+security privileges necessary for monitoring your node, as per the [official documentation](https://github.com/percona/mongodb_exporter#permissions).
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/prometheus.exporter.mongodb.md:27]:
```suggestion
We strongly recommend configuring a separate user for the Grafana Agent, giving it only the strictly mandatory security privileges necessary for monitoring your node. 
Refer to the [Percona documentation](https://github.com/percona/mongodb_exporter#permissions) for more information.
```

**Review by clayton-cornell — COMMENTED** [review]
Some minor suggestions to the docs

File: docs/sources/flow/reference/components/prometheus.exporter.mongodb.md (hunk 1 above)
Comment (gaantunes) [reply docs/sources/flow/reference/components/prometheus.exporter.mongodb.md:99]: Yes, this is still valid. Should we put integration specific information inside the agent component docs, though? 🤔

File: docs/sources/flow/reference/components/prometheus.exporter.mongodb.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`mongodb_uri` | `string` | MongoDB node connection URI. | | Yes
+
+MongoDB node connection URI must be in the [`Standard Connection String Format`](https://docs.mongodb.com/manual/reference/connection-string/#std-label-connections-standard-connection-string-format)
+
+
+
+## Exported fields
```
Comment (gaantunes) [inline docs/sources/flow/reference/components/prometheus.exporter.mongodb.md:49]: There was a request from a customer today, to enable the use of TLS certs to connect to mongodb. It is supported on the upstream exporter, so maybe we should extrapolate those? https://raintank-corp.slack.com/archives/C01343TNHCM/p1688732477656349

File: docs/sources/flow/reference/components/prometheus.exporter.mongodb.md (hunk 1 above)
Comment (marctc) [reply docs/sources/flow/reference/components/prometheus.exporter.mongodb.md:99]: You are right, removing then

File: docs/sources/flow/reference/components/prometheus.exporter.mongodb.md (hunk 6 above)
Comment (marctc) [reply docs/sources/flow/reference/components/prometheus.exporter.mongodb.md:49]: Should we do it in a follow up PR? I don't have much context of this exporter, just porting things to Flow.

**Pull request merged**